test/functional/api/cas/core.py

```diff
@@ -36,16 +36,17 @@ def __init__(self, core_device: str, cache_id: int):
         self.partitions = []
         self.block_size = None
 
-    def __get_core_info(self):
+    def __get_core_info(self, core_id=None):
         output = TestRun.executor.run(
             list_cmd(OutputFormat.csv.name, by_id_path=False))
         if output.exit_code != 0:
             raise Exception("Failed to execute list caches command.")
         output_lines = output.stdout.splitlines()
         for line in output_lines:
             split_line = line.split(',')
+            core_dev_id = self.core_device.get_device_id() if core_id is None else core_id
             if split_line[0] == "core" and (
-                    split_line[2] == os.path.join("/dev", self.core_device.get_device_id())
+                    split_line[2] == os.path.join("/dev", core_dev_id)
                     or split_line[5] == self.path):
```

Comment:
Mixing a parameter passed as an argument relating to a core and **self**, which also relates to a core (not necessarily the same) in one check is confusing.